### PR TITLE
Weather App - Ylva Nilsson

### PR DESCRIPTION
**Pull request by ylvan75**
**Link to my deployed project:**
[https://weather-app-by-ylva.netlify.app/](https://weather-app-by-ylva.netlify.app/)

**My reflections on how this week's project turned out:**
I'm proud of how we've worked together this week. Felt nice that we had most of the Javascript
part in place and basic design early in the week.

I think the structure of the JS-file feels good and I hope the code is easy to read.

**Things I'd like to have clarified or explained in more detail:**
I would have liked to create a function that takes search city and return with timezone that city shall have
and use the timezone value in the `Date()` object.

### CODE REVIEW

File: code/index.html [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          <label for="cityNamePicked" class="city-serch"
+            >Look up the weather for favourite city</label
+          >
+          <input
+            type="text"
+            id="cityNamePicked"
+            name="ciyNamePicked"
+            class="city-name-picked"
+            onkeydown="if (event.keyCode == 13)
+                document.getElementById('btnSearch').click()"
```
Comment (trangtara) [inline code/index.html:38]: This is new to me. So great!!!!

File: code/index.html [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            class="search-btn"
+            type="button"
+            id="btnSearch"
+            value="Search"
+            onclick="citySelected();"
+          />
+        </div>
+      </div>
+    </section>
+    <script src="weather.js"></script>
```
Comment (trangtara) [inline code/index.html:51]: Super clean HTML file. Well done!

File: code/style.css [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+.location {
+  font-size: 24px;
+  font-weight: 600;
+}
+
+.description::first-letter {
+  text-transform: capitalize;
+}
```
Comment (trangtara) [inline code/style.css:47]: Super! Simple and efficient code to capitalize the first letter of the description.

File: code/style.css [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+img {
+  height: 50px;
+  width: 50px;
+}
+
+.temp-forcast {
+  width: 100px;
+  text-align: end;
```
Comment (trangtara) [inline code/style.css:75]: cool with text-align: end.

File: code/style.css [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+/****MEDIA FOR DESKTOP *****/
+@media (min-width: 1024px) {
+  .weather-wrapper {
+    width: 600px;
+  }
+
+  .day {
+    width: 280px;
+  }
+}
```
Comment (trangtara) [inline code/style.css:138]: css codes are clean and consistent. Love it!!!

File: code/weather.js [hunk 6]
```diff
@@ -0,0 +1,145 @@
+const containerWeather = document.getElementById("weather");
+const containerToday = document.getElementById("weatherToday");
+const containerForecast = document.getElementById("weatherForecast");
+let citySearched = "Ösmo";
+
```
Comment (trangtara) [inline code/weather.js:5]: I like the way to set the default page.

File: code/weather.js [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+  const sunset = calculatingTime(weatherToday.sys.sunset);
+  const iconToday = iconDependingOnWeather(weatherToday.weather[0].main);
+  const description = weatherToday.weather[0].description;
+  weatherBackgroundColor(weatherToday.main.temp);
+
+  // Separate and build up the HTML tree
+  let weatherTodayHTML = "";
+  weatherTodayHTML += ` <img src='${iconToday}'>`;
+  weatherTodayHTML += `<div class="location-information">`;
+  weatherTodayHTML += `<p class="temp-today"> ${temperature}\xB0 </p>`;
```
Comment (trangtara) [inline code/weather.js:35]: Wow, I learned one new thing here about xB0.

File: code/weather.js [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+  weatherTodayHTML += `<p class="description"> ${description} </p>`;
+  weatherTodayHTML += `<p class="local-time"> ${timeInCity} </p>`;
+  weatherTodayHTML += `</div>`;
+
+  weatherTodayHTML += `<div class="sun-information">`;
+  weatherTodayHTML += `<p class="sunrise"> Sunrise ${sunrise}</p>`;
+  weatherTodayHTML += `<p class="sunset"> Sunset ${sunset}</p>`;
+  weatherTodayHTML += `</div>`;
+  return weatherTodayHTML;
+};
```
Comment (trangtara) [inline code/weather.js:46]: Simple and effective. Nice job!

File: code/weather.js [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+    .then((weatherForcast) => {
+      containerForecast.innerHTML = "";
+      const filteredForcast = weatherForcast.list.filter((item) =>
+        item.dt_txt.includes("12:00")
+      );
+
+      filteredForcast.forEach((forcast) => {
+        containerForecast.innerHTML += generatedHTMLForWeatherForcast(forcast);
+      });
+    });
```
Comment (trangtara) [inline code/weather.js:65]: Very clean code.

File: code/weather.js [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+const calculatedTemperature = (number) => {
+  const roundedTemp = Math.round(number * 10) / 10;
+  return roundedTemp;
+};
+
+// Function to convert time to a readable format
+const calculatingTime = (time) => {
+  const timeSet = new Date(time * 1000);
+  const timeString = timeSet.toLocaleTimeString("en-US", {
+    hour: "2-digit",
```
Comment (trangtara) [inline code/weather.js:96]: I like the way you separate functions into different sections, which makes your code very easy to follow. The note/comments are very helpful to read the code too.

File: code/weather.js [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+};
+
+// Function to  change background color depending on temperature
+const weatherBackgroundColor = (temp) => {
+  if (temp < 6) {
+    containerWeather.style.backgroundColor = "#b3cde3";
+  } else if (temp >= 6 && temp < 21) {
+    containerWeather.style.backgroundColor = "#fed98e";
+  } else containerWeather.style.backgroundColor = "#fe992a";
+};
```
Comment (trangtara) [inline code/weather.js:137]: Nice design! changing background according to the temperature.

**Review by trangtara — COMMENTED** [review]
Hi Ylva, I had a nice experience while reviewing your code and did learn a lot from you. Simple and clean codes on all HTML, css and JS.

File: code/weather.js [hunk 12]
```diff
@@ -0,0 +1,145 @@
...
+
+// Function invoked when search button is clicked
+const citySelected = () => {
+  citySearched = document.getElementById("cityNamePicked").value;
+  fetchWeatherForcast(citySearched);
+  fetchWeatherToday(citySearched);
+  document.getElementById("cityNamePicked").value = ""; //to clear input value after search
+};
```
Comment (trangtara) [inline code/weather.js:145]: Maybe a message/alert in case that users type in a city that does not exist will be helpful here.

**Review by trangtara — APPROVED** [review]
I just add a suggestion to the search function. It's not the requirement of the project, just a nice thing to do.

File: code/index.html [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <meta name="author" content="Ylva Nilsson" />
+    <title>Weather Forcast</title>
+    <link
+      href="https://fonts.googleapis.com/css2?family=Quicksand:wght@400;500;600;700;800&display=swap"
+      rel="stylesheet"
+    />
+    <link rel="stylesheet" href="style.css" />
```
Comment (EmOlsz) [inline code/index.html:13]: `href="./style.css"`

File: code/index.html [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+  </head>
+  <body>
+    <section id="weather" class="weather-container">
+      <div class="weather-wrapper">
+        <div id="weatherToday" class="weather-today-wrapper"></div>
+        <div id="weatherForecast" class="weather-forcast-wrapper"></div>
+
+        <div class="select-city">
+          <label for="cityNamePicked" class="city-serch"
+            >Look up the weather for favourite city</label
```
Comment (EmOlsz) [inline code/index.html:30]: Wrong line breaks

File: code/index.html [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+            onkeydown="if (event.keyCode == 13)
+                document.getElementById('btnSearch').click()"
+          />
+
+          <input
+            class="search-btn"
+            type="button"
+            id="btnSearch"
+            value="Search"
+            onclick="citySelected();"
```
Comment (EmOlsz) [inline code/index.html:46]: No need for semicolon here

File: code/index.html (hunk 2 above)
Comment (EmOlsz) [inline code/index.html:51]: `src="./weather.js"`

File: code/style.css [hunk 16]
```diff
@@ -0,0 +1,138 @@
...
+
+.weather-container {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+}
+
+.weather-wrapper {
+  margin: 0px 10px;
```
Comment (EmOlsz) [inline code/style.css:17]: We don't have to write `px` if value is 0, `margin: 0 10px;` is completely fine :)

**Review by EmOlsz — CHANGES_REQUESTED** [review]
Good job Ylva! I like how consistent your code and flow of it is. Your strategy for executing functions is well done :)
There are some minor changes I would like to see, take a look on my comments and provide them. After that, I'll approve your project

File: code/index.html (hunk 13 above)
Comment (ylvan75) [reply code/index.html]: Fixed!

Comment (ylvan75) [reply code/index.html]: Fixed!

File: code/index.html (hunk 2 above)
Comment (ylvan75) [reply code/index.html]: Fixed!

File: code/style.css (hunk 16 above)
Comment (ylvan75) [reply code/style.css]: Fixed!

File: code/style.css [hunk 17]
```diff
@@ -0,0 +1,138 @@
...
+  justify-content: space-evenly;
+}
+
+/*additional img styling to seperate from other img*/
+.weather-today-wrapper img {
+  position: absolute;
+  height: 150px;
+  width: 150px;
+  right: -10px;
+  top: 0px;
```
[the remark this answers is not in the document]
Comment (ylvan75) [reply code/style.css]: Fixed!

File: code/weather.js [hunk 18]
```diff
@@ -0,0 +1,145 @@
...
+  weatherForcast += `<img src='${iconForcast}'>`;
+  weatherForcast += `<p class="temp-forcast">${dailyTemp}\xB0/ ${tempFeelsLike}\xB0</p>`;
+  weatherForcast += `</div>`;
+  return weatherForcast;
+};
+
+// Function for temp rounded to one decimal
+const calculatedTemperature = (number) => {
+  const roundedTemp = Math.round(number * 10) / 10;
+  return roundedTemp;
```
[the remark this answers is not in the document]
Comment (ylvan75) [reply code/weather.js]: Fixed!

File: code/index.html (hunk 1 above)
Comment (ylvan75) [reply code/index.html]: @EmOlsz I have now placed this part in the .js. Perhaps I could have done the code nicer. What does it mean when I write input.addEventListner(... )  ?

Comment (EmOlsz) [reply code/index.html]: This method attaches new "action to be done" to specific event (click, key push, focus, mouse move etc.). It's just an alternative to writing `onclick`, `onmousemove` etc. directly in HTML

**Review by EmOlsz — APPROVED** [review]

**Pull request closed without merging**